python/mxnet/ndarray/sparse.py

```diff
@@ -711,47 +719,88 @@ def copyto(self, other):
             raise TypeError('copyto does not support type ' + str(type(other)))
 
 
-def _prepare_src_array(src, dtype, default_dtype):
-    """Prepare `src` and its dtype so that they can be used to construct NDArray.
-    `src` is converted to a `np.ndarray` if it's neither an `NDArray` nor an `np.ndarray`.
+def _prepare_src_array(source_array, dtype):
+    """Prepare `source_array` so that it can be used to construct NDArray.
+    `source_array` is converted to a `np.ndarray` if it's neither an `NDArray` \
+    nor an `np.ndarray`.
     """
-    if isinstance(src, NDArray):
-        dtype = src.dtype if dtype is None else dtype
-    else:
-        dtype = default_dtype if dtype is None else dtype
-        if not isinstance(src, np.ndarray):
-            try:
-                src = np.array(src, dtype=dtype)
-            except:
-                raise TypeError('values must be array like object')
-    return src, dtype
-
-
-def csr_matrix(data, indptr, indices, shape, ctx=None, dtype=None, indptr_type=None,
-               indices_type=None):
-    """Creates a 2D array with compressed sparse row (CSR) format.
+    if not isinstance(source_array, NDArray) and not isinstance(source_array, np.ndarray):
+        try:
+            source_array = np.array(source_array, dtype=dtype)
+        except:
+            raise TypeError('values must be array like object')
+    return source_array
+
+def _check_shape(s1, s2):
+    """check s1 == s2 if both are not None"""
+    if s1 and s2 and s1 != s2:
+        raise ValueError("Shape mismatch detected. " + str(s1) + " v.s. " + str(s2))
+
+def csr_matrix(arg1, shape=None, ctx=None, dtype=None):
+    """Creates a `CSRNDArray`, an 2D array with compressed sparse row (CSR) format.
+
+    The CSRNDArray can be instantiated in several ways:
+
+    - csr_matrix(D):
```

Comment:
Have you checked if this will render correctly?

Reply:
Yes the rendering is correct. I'll post a preview link soon